ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js

```diff
@@ -1,52 +1,129 @@
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { useSelector } from 'react-redux';
-import { getWeightedPermissions } from '../../../../helpers/utils/permission';
 import { useI18nContext } from '../../../../hooks/useI18nContext';
-import { Box } from '../../../component-library';
+import { Box, ButtonLink } from '../../../component-library';
 import {
   getMultipleTargetsSubjectMetadata,
   getSnapsMetadata,
 } from '../../../../selectors';
+import {
+  Display,
+  FlexDirection,
+  JustifyContent,
+} from '../../../../helpers/constants/design-system';
+import SnapPermissionAdapter from '../snap-permission-adapter';
+import { PermissionWeightThreshold } from '../../../../../shared/constants/permissions';
 import { getSnapName } from '../../../../helpers/utils/util';
-import SnapPermissionCell from '../snap-permission-cell';
+import { getWeightedPermissions } from '../../../../helpers/utils/permission';
 
 export default function SnapPermissionsList({
   snapId,
   snapName,
   permissions,
   connections,
   showOptions,
+  turnOffAbstraction,
+  showAllPermissions,
 }) {
   const t = useI18nContext();
-  const snapsMetadata = useSelector(getSnapsMetadata);
-  const permissionsToShow = {
-    ...permissions,
-    connection_permission: connections ?? {},
-  };
+
+  const combinedPermissions = useMemo(() => {
```

Comment:
why the need for `useMemo` now?

Reply:
Sorry I could have worded my confusion a bit better: was performance not an issue before? Where's the re-rendering coming from, this seems like it would mostly be a static component (when would permissions/connections change?) I asked about the needing to memoize this since we weren't doing it before.

Reply:
It is needed now because of dynamic requirements around permission set. Because `useEffect` is introduced and `combinedPermissions` are its dependency. I explained why in https://github.com/MetaMask/metamask-extension/pull/25175#discussion_r1653337073